smarty/sampler.py

```diff
@@ -332,38 +335,56 @@ def show_type_matches(self, atom_type_matches):
         return fraction_matched_atoms
     
     
-    def AtomDecorator(self, atom1type, decorators, natombasetypes):
-        """
-        Take a specified atomtype apply one of the provided decorators to it, returning a tuple containing a 
-        SMARTS string corresponding to a decorated atomtype and the corresponding name
-        """
-        if re.match('\$\(\*[=~:\-#](\w+)\)', decorators[0]) != None:
-            # There is a bond - two atom types
-            result = re.match('\[(.+)\]', atom1type[0])
-            basetype_index = random.randint(0, natombasetypes-1)
-            (basetype, basetype_typename) = self.atom_basetype[basetype_index]
-            new_dec = decorators[0].replace("z", basetype)
-            proposed_atomtype = '[' + result.groups(1)[0] + new_dec + ']'
-            proposed_typename = atom1type[1] + ' ' + basetype_typename +  ' ' + decorators[1]
+    #def AtomDecorator(self, atom1type, decorators, natombasetypes):
+    #    """
+    #    Take a specified atomtype apply one of the provided decorators to it, returning a tuple containing a 
+    #    SMARTS string corresponding to a decorated atomtype and the corresponding name
+    #    """
+    #    if re.match('\$\(\*[=~:\-#](\w+)\)', decorators[0]) != None:
+    #        # There is a bond - two atom types
+    #       result = re.match('\[(.+)\]', atom1type[0])
+    #        basetype_index = random.randint(0, natombasetypes-1)
+    #        (basetype, basetype_typename) = self.atom_basetype[basetype_index]
+    #        new_dec = decorators[0].replace("z", basetype)
+    #        proposed_atomtype = '[' + result.groups(1)[0] + new_dec + ']'
+    #        proposed_typename = atom1type[1] + ' ' + basetype_typename +  ' ' + decorators[1]
+    #    else:
+    #        # Only one atom type
+    #        result = re.match('\[(.+)\]', atom1type[0])
+    #        proposed_atomtype = '[' + result.groups(1)[0] + decorators[0] + ']'
+    #        proposed_typename = atom1type[1] + ' ' + decorators[1]
+    #    return proposed_atomtype, proposed_typename
+
+
+    # I didn't understand why there were base types being introduced in this metho
+    # It should just be decorating the atom, nothing else. 
+    def AtomDecorator(self, atom1type, decorator):
```

Comment:
AtomDecorator should only add decorators, not substituents. I think this can be done more generally than we were before so I've updated this method just add a decorator to an atom with the correct syntax.